### ISSUE TITLE
Validation fails on uneven indentation

### ISSUE THREAD

**Issue by sanathkr** [awslabs/aws-sam-local#19]
Running on this template - https://github.com/awslabs/serverless-application-model/blob/master/examples/2016-10-31/inline_swagger/template.yaml, `sam validate` command panics at [line-numbers.go](https://github.com/awslabs/aws-sam-local/blob/develop/vendor/github.com/awslabs/goformation/resources/line-numbers.go#L116)

```
2017/08/11 12:08:53 0.1.0
2017/08/11 12:08:53 PANIC: YAML contains malformed structures
panic: PANIC: YAML contains malformed structures

goroutine 1 [running]:
log.Panicf(0x168d237, 0x29, 0x0, 0x0, 0x0)
	/home/ubuntu/bin/go/src/log/log.go:329 +0xda
github.com/awslabs/aws-sam-local/vendor/github.com/awslabs/goformation/resources.ProcessLineNumbers(0xc420597000, 0x48b, 0xe00, 0x168c1c0, 0x27)
	/home/ubuntu/repo/src/github.com/awslabs/aws-sam-local/vendor/github.com/awslabs/goformation/resources/line-numbers.go:116 +0xe8f
github.com/awslabs/aws-sam-local/vendor/github.com/awslabs/goformation.unmarshal(0xc420597000, 0x48b, 0xe00, 0x1680cef, 0x16, 0x0, 0x0)
	/home/ubuntu/repo/src/github.com/awslabs/aws-sam-local/vendor/github.com/awslabs/goformation/unmarshal.go:45 +0x403
github.com/awslabs/aws-sam-local/vendor/github.com/awslabs/goformation.Parse(0xc420597000, 0x48b, 0xe00, 0x48b, 0xe00, 0x0, 0x0, 0xc4204c8028, 0x0)
	/home/ubuntu/repo/src/github.com/awslabs/aws-sam-local/vendor/github.com/awslabs/goformation/goformation.go:49 +0xfe
github.com/awslabs/aws-sam-local/vendor/github.com/awslabs/goformation.read(0x1966360, 0xc4204c8028, 0xc4204c8028, 0x0, 0x0, 0x0, 0x1, 0x1)
	/home/ubuntu/repo/src/github.com/awslabs/aws-sam-local/vendor/github.com/awslabs/goformation/goformation.go:40 +0x105
github.com/awslabs/aws-sam-local/vendor/github.com/awslabs/goformation.Open(0xc42007e467, 0xd, 0xc42007e467, 0xd, 0x13, 0x0, 0x0, 0xc420220f90)
	/home/ubuntu/repo/src/github.com/awslabs/aws-sam-local/vendor/github.com/awslabs/goformation/goformation.go:29 +0x27e
main.validate(0xc4203ce780)
	/home/ubuntu/repo/src/github.com/awslabs/aws-sam-local/validate.go:13 +0x7a
github.com/awslabs/aws-sam-local/vendor/github.com/codegangsta/cli.HandleAction(0x1587e60, 0x169a720, 0xc4203ce780, 0xc4204b8a00, 0x0)
	/home/ubuntu/repo/src/github.com/awslabs/aws-sam-local/vendor/github.com/codegangsta/cli/app.go:487 +0x7c
github.com/awslabs/aws-sam-local/vendor/github.com/codegangsta/cli.Command.Run(0x1677894, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x16971ee, 0xa7, 0x0, ...)
	/home/ubuntu/repo/src/github.com/awslabs/aws-sam-local/vendor/github.com/codegangsta/cli/command.go:207 +0xb72
github.com/awslabs/aws-sam-local/vendor/github.com/codegangsta/cli.(*App).Run(0xc420154680, 0xc42000a1e0, 0x2, 0x2, 0x0, 0x0)
	/home/ubuntu/repo/src/github.com/awslabs/aws-sam-local/vendor/github.com/codegangsta/cli/app.go:250 +0x7d0
main.main()
	/home/ubuntu/repo/src/github.com/awslabs/aws-sam-local/main.go:302 +0x1be1
```

**Comment by MCLDG** [comment]
I'm having a similar problem using a SAM template that I've been using for months. Based on the error message its difficult to know where to start looking for the problem. I'd also suggest that if a template is readily accepted by 'aws cloudformation package', it should probably be accepted by SAM local:

sam validate --template sam-airmile-with-stack-dependency.yml 
2017/08/15 10:05:47 0.1.0
ERROR: Failed to read the resources line numbers. This usually means that the template contains invalid indentation. Please check it and try again

**Comment by sanathkr** [comment]
Yup. That's an unfortunate, but known bug, which we already have a fix for in the awslabs/goformation repository. I will try to get it merged into this asap. The template is most certainly valid, but goformation, the package that tries to validate templates, fails to understand unequal indentation.

**Comment by sanathkr** [comment]
@pesama how hard is it to remove line number checking altogether?

**Comment by PaulMaddox** [comment]
Can confirm this is fixed in goformation 0.1.0 branch.

**Comment by sanathkr** [comment]
Fixed in [v0.2.0](https://github.com/awslabs/aws-sam-local/releases/tag/v0.2.0)
